File: scripts/old/naming_helpers.py
"""
naming_helpers.py
Utilities for generating standardized output filenames
based on naming rules in config.yml.
"""

from datetime import date
from typing import Iterable, Union

def _date_range_string(start: Union[str, date], end: Union[str, date]) -> str:
    if isinstance(start, date):
        start = start.strftime("%Y%m%d")
    if isinstance(end, date):
        end = end.strftime("%Y%m%d")
    return f"{start}-{end}"

def name_single_site(site: Union[int, str], test: str,
                     start_date: Union[str, date],
                     end_date: Union[str, date],
                     ext: str = "csv") -> str:
    site_str = str(site)
    dr = _date_range_string(start_date, end_date)
    fname = f"site_{site_str}_{test}_{dr}"
    return f"{fname}.{ext}"

def name_single_site_aggregated(site: Union[int, str], test: str, agg: str,
                                start_date: Union[str, date],
                                end_date: Union[str, date],
                                ext: str = "csv") -> str:
    site_str = str(site)
    dr = _date_range_string(start_date, end_date)
    fname = f"site_{site_str}_{test}_{agg}_{dr}"
    return f"{fname}.{ext}"

def name_multi_site(sites: Iterable[Union[int, str]], test: str,
                    start_date: Union[str, date],
                    end_date: Union[str, date],
                    ext: str = "csv") -> str:
    site_list = "_".join(str(s) for s in sites)
    dr = _date_range_string(start_date, end_date)
    fname = f"sites_{site_list}_{test}_{dr}"
    return f"{fname}.{ext}"

def name_all_sites(test: str,
                   start_date: Union[str, date],
                   end_date: Union[str, date],
                   ext: str = "csv") -> str:
    dr = _date_range_string(start_date, end_date)
    fname = f"all_{test}_{dr}"
    return f"{fname}.{ext}"

File: scripts/old/naming_helpers.R
# naming_helpers.R
# Utilities for generating standardized output filenames
# using the naming rules in config.yml

if (!exists("cfg")) {
  if (file.exists("config.yml")) {
    cfg <- yaml::read_yaml("config.yml")
  } else {
    stop("config.yml not found; cannot load naming rules.")
  }
}

.date_range_string <- function(start, end) {
  # start, end: character "YYYYMMDD" or Date objects
  if (inherits(start, "Date")) start <- format(start, "%Y%m%d")
  if (inherits(end, "Date"))   end   <- format(end, "%Y%m%d")
  paste0(start, "-", end)
}

name_single_site <- function(site, test, start_date, end_date, ext = "csv") {
  site <- as.character(site)
  dr <- .date_range_string(start_date, end_date)
  fname <- paste0("site_", site, "_", test, "_", dr)
  paste0(fname, ".", ext)
}

name_single_site_aggregated <- function(site, test, agg, start_date, end_date, ext = "csv") {
  site <- as.character(site)
  dr <- .date_range_string(start_date, end_date)
  fname <- paste0("site_", site, "_", test, "_", agg, "_", dr)
  paste0(fname, ".", ext)
}

name_multi_site <- function(sites, test, start_date, end_date, ext = "csv") {
  # sites: vector of site IDs (numeric or character)
  sites <- as.character(sites)
  site_part <- paste(sites, collapse = "_")
  dr <- .date_range_string(start_date, end_date)
  fname <- paste0("sites_", site_part, "_", test, "_", dr)
  paste0(fname, ".", ext)
}

name_all_sites <- function(test, start_date, end_date, ext = "csv") {
  dr <- .date_range_string(start_date, end_date)
  fname <- paste0("all_", test, "_", dr)
  paste0(fname, ".", ext)
}
